### DISCUSSION TITLE
How to get non-reply posts for a fixed list of hashtags?

### DISCUSSION THREAD

**Discussion by nileshtrivedi** [neet/masto.js#1154] [Q&A]
I am trying to use hashtags to simulate Slack's channels. While there's an API get a list of trending tags, I need to build a page (in Elk) showing 10+ posts under each of the hashtag from a list curated by the admin. How can this be achieved?

Please [see this](https://github.com/elk-zone/elk/discussions/2923) for some additional context.

**Comment by neet** [comment]
Hello @nileshtrivedi. We have a `/timeline` API to look for posts containing a specific hashtag.
https://docs.joinmastodon.org/methods/timelines/#tag

I hope this helps

**Reply by nileshtrivedi** [reply]
Thanks @neet . So if i want to show 5 posts for each of the 30 hashtags on the same page, I would have to make 30 calls to `/timeline`? This could be problematic.

**Reply by nileshtrivedi** [reply]
I did a bit of digging and submitted this feature request to Mastodon: https://github.com/mastodon/mastodon/issues/31505

**Comment by michelcrypt4d4mus** [comment]
> _So if i want to show 5 posts for each of the 30 hashtags on the same page, I would have to make 30 calls to /timeline?_

fwiw in [FediAlgo](https://github.com/neet/masto.js/discussions/1310) we just do exactly this using a semaphore to slightly slow down/space out the requests, but for more than 30 tags (~80) and it's pretty much fine.
